README.md

```diff
@@ -0,0 +1,238 @@
+
+    ![Leases](media/cosmos_leases.JPG)
+
+1. Press enter again in the terminal. This will trigger 10 documents to be inserted into **InventoryContainer**. Each document insertion appears in the Change Feed as JSON; the following callback code handles these events by mirroring the JSON documents into a materialized view:
+
+    **Java SDK 4.0**
+    ```java
+    public static ChangeFeedProcessor getChangeFeedProcessor(String hostName, CosmosAsyncContainer feedContainer, CosmosAsyncContainer leaseContainer) {
+        ChangeFeedProcessorOptions cfOptions = new ChangeFeedProcessorOptions();
+        cfOptions.setFeedPollDelay(Duration.ofMillis(100));
+        cfOptions.setStartFromBeginning(true);
+        return ChangeFeedProcessor.changeFeedProcessorBuilder()
+            .setOptions(cfOptions)
+            .setHostName(hostName)
+            .setFeedContainer(feedContainer)
+            .setLeaseContainer(leaseContainer)
+            .setHandleChanges((List<JsonNode> docs) -> {
+                for (JsonNode document : docs) {
+                        //Duplicate each document update from the feed container into the materialized view container
+                        updateInventoryTypeMaterializedView(document);
+                }
+
+            })
+            .build();
+    }
+
+    private static void updateInventoryTypeMaterializedView(JsonNode document) {
+        typeContainer.upsertItem(document).subscribe();
+    }
+    ```
+
+    **Java SDK 3.7.0**
+    ```java
+    public static ChangeFeedProcessor getChangeFeedProcessor(String hostName, CosmosContainer feedContainer, CosmosContainer leaseContainer) {
+        ChangeFeedProcessorOptions cfOptions = new ChangeFeedProcessorOptions();
+        cfOptions.feedPollDelay(Duration.ofMillis(100));
+        cfOptions.startFromBeginning(true);
+        return ChangeFeedProcessor.Builder()
+            .options(cfOptions)
+            .hostName(hostName)
+            .feedContainer(feedContainer)
+            .leaseContainer(leaseContainer)
+            .handleChanges((List<CosmosItemProperties> docs) -> {
+                for (CosmosItemProperties document : docs) {
+                        //Duplicate each document update from the feed container into the materialized view container
+                        updateInventoryTypeMaterializedView(document);
+                }
+
+            })
+            .build();
+    }
+
+    private static void updateInventoryTypeMaterializedView(CosmosItemProperties document) {
+        typeContainer.upsertItem(document).subscribe();
+    }    
+    ```
+
+1. Allow the code to run 5-10sec. Then return to the Azure Portal Data Explorer and navigate to **InventoryContainer > items**. You should see that items are being inserted into the inventory container; note the partition key (```id```).
```

Comment:
/id in general is a bad partition key; for this example /brand might be a much better choice. we will need to iterate picking a random brand from a fixed local list of brands for each document being inserted...

Reply:
Hmm... I think I understand where /id comes from. Still I think it's still not the best use for a partition key. Having a brand list with only couple entries might still be a better choice.